src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/DefaultProcessConfigurationTests.cs

```diff
@@ -0,0 +1,155 @@
+                Value = "arg1"
+            };
+
+            var filter = CreateFilterEntry(filterDescriptorPid);
+            ValidateProcessFilter(filter, DiagProcessFilterCriteria.ProcessId, filterDescriptorPid.Value);
+
+            filter = CreateFilterEntry(filterDescriptorName);
+            ValidateProcessFilter(filter, DiagProcessFilterCriteria.ProcessName, filterDescriptorName.Value);
+
+            filter = CreateFilterEntry(filterDescriptorNameContains);
+            ValidateProcessFilter(filter, DiagProcessFilterCriteria.ProcessName, filterDescriptorNameContains.Value, DiagProcessFilterMatchType.Contains);
+
+            filter = CreateFilterEntry(filterDescriptorCommand);
+            ValidateProcessFilter(filter, DiagProcessFilterCriteria.CommandLine, filterDescriptorCommand.Value);
+
+            filter = CreateFilterEntry(filterDescriptorCommandContains);
+            ValidateProcessFilter(filter, DiagProcessFilterCriteria.CommandLine, filterDescriptorCommandContains.Value, DiagProcessFilterMatchType.Contains);
+
+            //This filter doesn't make any sense but we are just testing that we can combine multiple filters
+            var options = CreateOptions(filterDescriptorPid, filterDescriptorName, filterDescriptorNameContains, filterDescriptorCommand, filterDescriptorCommandContains);
+
+            ValidateProcessFilter(options.Filters[0], DiagProcessFilterCriteria.ProcessId, filterDescriptorPid.Value);
+            ValidateProcessFilter(options.Filters[1], DiagProcessFilterCriteria.ProcessName, filterDescriptorName.Value);
+            ValidateProcessFilter(options.Filters[2], DiagProcessFilterCriteria.ProcessName, filterDescriptorNameContains.Value, DiagProcessFilterMatchType.Contains);
+            ValidateProcessFilter(options.Filters[3], DiagProcessFilterCriteria.CommandLine, filterDescriptorCommand.Value);
+            ValidateProcessFilter(options.Filters[4], DiagProcessFilterCriteria.CommandLine, filterDescriptorCommandContains.Value, DiagProcessFilterMatchType.Contains);
+        }
+
+        [Fact]
+        public void NewCriteriaTest()
+        {
+            //When new enumerations (such as Entrypoint) are added to DiagProcessFilterCriteria,
+            //this test will fail. This means new unit tests should be written for that criteria and this test can be updated.
+
+            //Sorted by unsigned integer value
+            var expectedValues = Enum.GetValues(typeof(DiagProcessFilterCriteria));
+
+            var actualValues = new[]
+            {
+                DiagProcessFilterCriteria.ProcessId,
+                DiagProcessFilterCriteria.RuntimeId,
+                DiagProcessFilterCriteria.ProcessName,
+                DiagProcessFilterCriteria.CommandLine
+            }.Cast<int>().ToArray();
+
+            Assert.Equal(expectedValues.Length, actualValues.Length);
+        }
+
+        private static void ValidateProcessFilter(DiagProcessFilterEntry filter,
```

Comment:
nit: rename from `filter` to `actualFilter`. Test verification methods typically differentiate between `expected` and `actual` values.